<!-- Spawn fake drones for testing algorithms-->

<launch>
  <arg name="fake_drone_cfg_filepath" default="$(find gestelt_bringup)/config/fake_drone_params.yaml"/>
  <arg name="map_width" default="11.5"/>
  <arg name="map_length" default="20.0"/>

  <arg name="map_origin_x" default="$(eval -arg('map_length')/2)"/>
  <arg name="map_origin_y" default="$(eval arg('map_width')/2)"/>

  <arg name="swarm_origin_x" default="$(eval arg('map_origin_x') - 0.5)"/>
  <arg name="swarm_origin_y" default="$(eval arg('map_origin_y') - 2.0)"/>

  <arg name="height" default="0.5"/>

  <include file="$(find gestelt_bringup)/launch/fake_drone/include/fake_drone_complete.xml">
      <arg name="fake_drone_cfg_filepath" value="$(arg fake_drone_cfg_filepath)"/>

      <arg name="drone_id" value="0"/>

      <arg name="init_x" value="$(eval arg('swarm_origin_x'))"/>
      <arg name="init_y" value="$(eval arg('swarm_origin_y') - 0*0.5)"/>
      <arg name="init_z" value="$(arg height)"/>
  </include>

  <include file="$(find gestelt_bringup)/launch/fake_drone/include/fake_drone_complete.xml">
      <arg name="fake_drone_cfg_filepath" value="$(arg fake_drone_cfg_filepath)"/>

      <arg name="drone_id" value="1"/>

      <arg name="init_x" value="$(eval arg('swarm_origin_x'))"/>
      <arg name="init_y" value="$(eval arg('swarm_origin_y') - 1*0.5)"/>
      <arg name="init_z" value="$(arg height)"/>
  </include>

  <include file="$(find gestelt_bringup)/launch/fake_drone/include/fake_drone_complete.xml">
      <arg name="fake_drone_cfg_filepath" value="$(arg fake_drone_cfg_filepath)"/>

      <arg name="drone_id" value="2"/>

      <arg name="init_x" value="$(eval arg('swarm_origin_x'))"/>
      <arg name="init_y" value="$(eval arg('swarm_origin_y') - 2*0.5)"/>
      <arg name="init_z" value="$(arg height)"/>
  </include>

  <include file="$(find gestelt_bringup)/launch/fake_drone/include/fake_drone_complete.xml">
      <arg name="fake_drone_cfg_filepath" value="$(arg fake_drone_cfg_filepath)"/>

      <arg name="drone_id" value="3"/>

      <arg name="init_x" value="$(eval arg('swarm_origin_x'))"/>
      <arg name="init_y" value="$(eval arg('swarm_origin_y') - 3*0.5)"/>
      <arg name="init_z" value="$(arg height)"/>
  </include>

  <include file="$(find gestelt_bringup)/launch/fake_drone/include/fake_drone_complete.xml">
      <arg name="fake_drone_cfg_filepath" value="$(arg fake_drone_cfg_filepath)"/>

      <arg name="drone_id" value="4"/>

      <arg name="init_x" value="$(eval arg('swarm_origin_x'))"/>
      <arg name="init_y" value="$(eval arg('swarm_origin_y') - 4*0.5)"/>
      <arg name="init_z" value="$(arg height)"/>
  </include>

  <include file="$(find gestelt_bringup)/launch/fake_drone/include/fake_drone_complete.xml">
      <arg name="fake_drone_cfg_filepath" value="$(arg fake_drone_cfg_filepath)"/>

      <arg name="drone_id" value="5"/>

      <arg name="init_x" value="$(eval arg('swarm_origin_x'))"/>
      <arg name="init_y" value="$(eval arg('swarm_origin_y') - 5*0.5)"/>
      <arg name="init_z" value="$(arg height)"/>
  </include>

  <include file="$(find gestelt_bringup)/launch/fake_drone/include/fake_drone_complete.xml">
      <arg name="fake_drone_cfg_filepath" value="$(arg fake_drone_cfg_filepath)"/>

      <arg name="drone_id" value="6"/>

      <arg name="init_x" value="$(eval arg('swarm_origin_x'))"/>
      <arg name="init_y" value="$(eval arg('swarm_origin_y') - 6*0.5)"/>
      <arg name="init_z" value="$(arg height)"/>
  </include>

  <include file="$(find gestelt_bringup)/launch/fake_drone/include/fake_drone_complete.xml">
      <arg name="fake_drone_cfg_filepath" value="$(arg fake_drone_cfg_filepath)"/>

      <arg name="drone_id" value="7"/>

      <arg name="init_x" value="$(eval arg('swarm_origin_x'))"/>
      <arg name="init_y" value="$(eval arg('swarm_origin_y') - 7*0.5)"/>
      <arg name="init_z" value="$(arg height)"/>
  </include>

  <include file="$(find gestelt_bringup)/launch/fake_drone/include/fake_drone_complete.xml">
      <arg name="fake_drone_cfg_filepath" value="$(arg fake_drone_cfg_filepath)"/>

      <arg name="drone_id" value="8"/>

      <arg name="init_x" value="$(eval arg('swarm_origin_x'))"/>
      <arg name="init_y" value="$(eval arg('swarm_origin_y') - 8*0.5)"/>
      <arg name="init_z" value="$(arg height)"/>
  </include>


  <include file="$(find gestelt_bringup)/launch/fake_drone/include/fake_drone_complete.xml">
      <arg name="fake_drone_cfg_filepath" value="$(arg fake_drone_cfg_filepath)"/>

      <arg name="drone_id" value="9"/>

      <arg name="init_x" value="$(eval arg('swarm_origin_x'))"/>
      <arg name="init_y" value="$(eval arg('swarm_origin_y') - 9*0.5)"/>
      <arg name="init_z" value="$(arg height)"/>
  </include>


  <include file="$(find gestelt_bringup)/launch/fake_drone/include/fake_drone_complete.xml">
      <arg name="fake_drone_cfg_filepath" value="$(arg fake_drone_cfg_filepath)"/>

      <arg name="drone_id" value="10"/>

      <arg name="init_x" value="$(eval arg('swarm_origin_x'))"/>
      <arg name="init_y" value="$(eval arg('swarm_origin_y') - 10*0.5)"/>
      <arg name="init_z" value="$(arg height)"/>
  </include>


  <include file="$(find gestelt_bringup)/launch/fake_drone/include/fake_drone_complete.xml">
      <arg name="fake_drone_cfg_filepath" value="$(arg fake_drone_cfg_filepath)"/>

      <arg name="drone_id" value="11"/>

      <arg name="init_x" value="$(eval arg('swarm_origin_x'))"/>
      <arg name="init_y" value="$(eval arg('swarm_origin_y') - 11*0.5)"/>
      <arg name="init_z" value="$(arg height)"/>
  </include>

  <include file="$(find gestelt_bringup)/launch/fake_drone/include/fake_drone_complete.xml">
      <arg name="fake_drone_cfg_filepath" value="$(arg fake_drone_cfg_filepath)"/>

      <arg name="drone_id" value="12"/>

      <arg name="init_x" value="$(eval arg('swarm_origin_x'))"/>
      <arg name="init_y" value="$(eval arg('swarm_origin_y') - 12*0.5)"/>
      <arg name="init_z" value="$(arg height)"/>
  </include>


  <include file="$(find gestelt_bringup)/launch/fake_drone/include/fake_drone_complete.xml">
      <arg name="fake_drone_cfg_filepath" value="$(arg fake_drone_cfg_filepath)"/>

      <arg name="drone_id" value="13"/>

      <arg name="init_x" value="$(eval arg('swarm_origin_x'))"/>
      <arg name="init_y" value="$(eval arg('swarm_origin_y') - 13*0.5)"/>
      <arg name="init_z" value="$(arg height)"/>
  </include>


  <include file="$(find gestelt_bringup)/launch/fake_drone/include/fake_drone_complete.xml">
      <arg name="fake_drone_cfg_filepath" value="$(arg fake_drone_cfg_filepath)"/>

      <arg name="drone_id" value="14"/>

      <arg name="init_x" value="$(eval arg('swarm_origin_x'))"/>
      <arg name="init_y" value="$(eval arg('swarm_origin_y') - 14*0.5)"/>
      <arg name="init_z" value="$(arg height)"/>
  </include>


  <include file="$(find gestelt_bringup)/launch/fake_drone/include/fake_drone_complete.xml">
      <arg name="fake_drone_cfg_filepath" value="$(arg fake_drone_cfg_filepath)"/>

      <arg name="drone_id" value="15"/>

      <arg name="init_x" value="$(eval arg('swarm_origin_x'))"/>
      <arg name="init_y" value="$(eval arg('swarm_origin_y') - 15*0.5)"/>
      <arg name="init_z" value="$(arg height)"/>
  </include>


</launch>
